src/bitcoin/OrdinalsClient.ts

```diff
@@ -45,6 +35,96 @@ export class OrdinalsClient {
   async estimateFee(blocks: number = 1): Promise<number> {
     return Math.max(1, blocks) * 10;
   }
+
+  // Added provider-like helper methods commonly expected by higher-level resolvers
+  // Minimal placeholder implementations suitable for unit testing
+
+  async getSatInfo(satoshi: string): Promise<{ inscription_ids: string[] }> {
+    const data = await this.fetchJson<any>(`/sat/${satoshi}`);
+    if (!data) return { inscription_ids: [] };
+    // Support both {inscription_ids: []} and {data: {inscription_ids: []}} shapes via fetchJson
+    const inscriptionIds = Array.isArray(data.inscription_ids) ? data.inscription_ids : [];
+    return { inscription_ids: inscriptionIds };
+  }
+
+  async resolveInscription(identifier: string): Promise<OrdinalsInscription | null> {
+    if (!identifier) return null;
+    const info = await this.fetchJson<any>(`/inscription/${identifier}`);
+    if (!info) return null;
+
+    // Fetch content bytes
+    const contentUrl = info.content_url || `${this.rpcUrl}/content/${identifier}`;
+    const contentRes = await fetch(contentUrl);
+    if (!contentRes.ok) throw new Error(`Failed to fetch inscription content: ${contentRes.status}`);
+    const contentArrayBuf = await contentRes.arrayBuffer();
+    const content = Buffer.from(new Uint8Array(contentArrayBuf));
+
+    // owner_output may be 'txid:vout'
+    let txid = 'unknown';
+    let vout = 0;
+    if (typeof info.owner_output === 'string' && info.owner_output.includes(':')) {
+      const [tid, v] = info.owner_output.split(':');
+      txid = tid;
+      vout = Number(v) || 0;
+    }
+
+    // sat number provided as number or string
+    const satoshi = String(info.sat ?? '');
+
+    const inscription: OrdinalsInscription = {
+      satoshi,
+      inscriptionId: info.inscription_id || identifier,
+      content,
+      contentType: info.content_type || 'application/octet-stream',
+      txid,
+      vout,
+      blockHeight: info.block_height
+    };
+
+    return inscription;
+  }
+
+  async getMetadata(inscriptionId: string): Promise<Record<string, unknown> | null> {
+    if (!inscriptionId) return null;
+    const info = await this.fetchJson<any>(`/inscription/${inscriptionId}`);
+    if (!info) return null;
+
+    // If API provides metadata directly
```

Comment:
@cursoragent it will not provide metada directly. This whole function is wrong maybe the ordinalsplus package has these mistakes too? The metadata endpoint will ALWAYS need to be used and that endpoint ALWAYS returns cbor